### PR TITLE
changes: note deprecated function/macros with no replacement.

### LINKED ISSUE #14616: Deprecated low level key API's that have no replacements
Whilst updating the documentation related to deprecation's I noticed the following had no replacements..
Should these be supported?

DH:

void DH_clear_flags(DH *dh, int flags);
 int DH_test_flags(const DH *dh, int flags);
 void DH_set_flags(DH *dh, int flags);

The flag DH_FLAG_CACHE_MONT_P is not accessible.
DH_FLAG_TYPE_DHX, DH_FLAG_TYPE_DH - EVP_is_a() for the test_flags, not sure set or get should be used.

DH *DH_get_1024_160(void);
DH *DH_get_2048_224(void);
DH *DH_get_2048_256(void);
Maybe these are not that useful - the DH gen handles these as RFC5114 types as a group name - I have updated the docs to reflect this.

DSA:
void DSA_clear_flags(DSA *d, int flags);
int DSA_test_flags(const DSA *d, int flags);
void DSA_set_flags(DSA *d, int flags);
DSA_FLAG_CACHE_MONT_P

DH *DSA_dup_DH(const DSA *r) - maybe not useful externally?
DSAparams_dup()

RSA:

void RSA_clear_flags(RSA *r, int flags);
int RSA_test_flags(const RSA *r, int flags);
void RSA_set_flags(RSA *r, int flags);
Flags
RSA_METHOD_FLAG_NO_CHECK
RSA_FLAG_CACHE_PUBLIC
RSA_FLAG_CACHE_PRIVATE
RSA_FLAG_BLINDING
RSA_FLAG_THREAD_SAFE - cant see this used..
RSA_FLAG_EXT_PKEY
RSA_FLAG_NO_BLINDING - related to the blinding functions below

RSA_get_version()

int RSA_blinding_on(RSA *rsa, BN_CTX *ctx);
void RSA_blinding_off(RSA *rsa);
BN_BLINDING *RSA_setup_blinding(RSA *rsa, BN_CTX *ctx) - I assume there will be no replacement for this.

RSAPrivateKey_dup()
RSAPublicKey_dup()

### PR DESCRIPTION
**Pull request by paulidale**
These functions are deprecated with no replacement specified:

    DH_clear_flags, DH_get_1024_160, DH_get_2048_224, DH_get_2048_256,
    DH_set_flags, DH_test_flags, DSA_clear_flags, DSA_dup_DH,
    DSAparams_dup, DSA_set_flags, DSA_test_flags, RSA_blinding_off,
    RSA_blinding_on, RSA_clear_flags, RSA_get_version, RSAPrivateKey_dup,
    RSAPublicKey_dup, RSA_set_flags, RSA_setup_blinding and
    RSA_test_flags.

The flags that are going are:

    DH_FLAG_CACHE_MONT_P, DSA_FLAG_CACHE_MONT_P,
    RSA_FLAG_BLINDING, RSA_FLAG_CACHE_PRIVATE, RSA_FLAG_CACHE_PUBLIC,
    RSA_FLAG_EXT_PKEY, RSA_FLAG_NO_BLINDING, RSA_FLAG_THREAD_SAFE and
    RSA_METHOD_FLAG_NO_CHECK.

These two flags are "readable" via EVP_is_a().  They are not writable:

    DH_FLAG_TYPE_DHX and DH_FLAG_TYPE_DH.

Fixes #14616

### CODE REVIEW

File: CHANGES.md [hunk 1]
```diff
@@ -756,6 +756,19 @@ OpenSSL 3.0
    time.  Instead applications should use L<EVP_PKEY_encrypt_init(3)>,
    L<EVP_PKEY_encrypt(3)>, L<EVP_PKEY_decrypt_init(3)> and
    L<EVP_PKEY_decrypt(3)>.
+   
+   All of theese low level RSA functions have been deprecated without
```
Comment (slontis) [inline CHANGES.md:760]: Maybe theeeeeeees ?

File: CHANGES.md [hunk 2]
```diff
@@ -847,6 +870,13 @@ OpenSSL 3.0
...
    L<EVP_DigestSignUpdate(3)> and L<EVP_DigestSignFinal(3)>.
 
+   These low level DSA functions have been deprecated without replacement:
+
+   DSA_clear_flags, DSA_dup_DH, DSAparams_dup, DSA_set_flags and
+   DSA_test_flags.
+
+   The DSA_FLAG_CACHE_MONT_P flag has been deprecated without replacement.
+
    Finaly functions that assign or obtain DSA objects from an EVP_PKEY such as
```
Comment (slontis) [inline CHANGES.md:880]: While you are here - change this to 'finally'

Comment (paulidale) [reply CHANGES.md:880]: Done, and a second instance of the same.

**Review by slontis — APPROVED** [review]

**Comment by paulidale** [comment]
Merged to master, thanks for the feedback.

**Pull request closed without merging**